llvm/lib/SYCLLowerIR/LowerWGLocalMemory.cpp

```diff
@@ -52,6 +54,32 @@ ModulePass *llvm::createSYCLLowerWGLocalMemoryLegacyPass() {
   return new SYCLLowerWGLocalMemoryLegacy();
 }
 
+static bool inlineAllocateLocalMemoryFunc(Module &M) {
+  Function *ALMFunc = M.getFunction(SYCL_ALLOCLOCALMEM_CALL);
+  if (!ALMFunc)
+    return false;
+
+  auto *Caller = cast<CallInst>(*ALMFunc->user_begin())->getFunction();
+  if (!Caller->hasFnAttribute(Attribute::AlwaysInline)) {
+    // Already inlined.
+    return false;
+  }
+  std::string FName = llvm::demangle(Caller->getName());
+  if (FName.find("sycl::_V1::ext::oneapi::group_local_memory") ==
+      std::string::npos) {
+    // Already inlined.
+    return false;
+  }
+  for (User *U : make_early_inc_range(Caller->users())) {
+    auto *CI = cast<CallInst>(U);
+    InlineFunctionInfo IFI;
+    [[maybe_unused]] auto Result = InlineFunction(*CI, IFI);
+    assert(Result.isSuccess() && "inlining failed");
```

Comment:
Should we handle the inline failure better? Or else if we get inline failure in release build, then we silently remove the call?

Reply:
the code here aligns with https://github.com/intel/llvm/blob/a9a68561c7ebbf8e9d21d6ce2c711d579262b185/llvm/lib/Transforms/Coroutines/CoroSplit.cpp#L323-L325
alwaysinline pass only emits remark when InlineFunction fails: https://github.com/intel/llvm/blob/a9a68561c7ebbf8e9d21d6ce2c711d579262b185/llvm/lib/Transforms/IPO/AlwaysInliner.cpp#L68-L78

Reply:
OK. Thanks.